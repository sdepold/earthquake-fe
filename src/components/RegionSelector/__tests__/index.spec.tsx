import { render } from "@testing-library/react";
import { RegionSelector, RegionSelectorProps } from "..";

jest.mock("../../Map", () => ({
  __esModule: true,
  default: () => <div>Map</div>,
}));

describe("EarthquakeListItem", () => {
  it("should match the expected HTML structure", () => {
    const props: RegionSelectorProps = {
      regions: [
        { name: "North America", slug: "north-america" },
        { name: "South America", slug: "south-america" },
      ],
      onChange: jest.fn(),
      value: "america",
    };

    const { container } = render(<RegionSelector {...props} />);
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent("North America");
  });
});
