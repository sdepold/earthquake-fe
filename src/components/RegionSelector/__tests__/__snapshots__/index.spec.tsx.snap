// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarthquakeListItem should match the expected HTML structure 1`] = `
<div>
  <div
    class="region-selector"
  >
    <span
      class="listbox-button"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        class="btn btn--form"
        style="width: 100%;"
        type="button"
      >
        <span
          class="btn__cell"
        >
          <span
            class="btn__floating-label"
          >
            Select a region
          </span>
          <span
            class="btn__text"
          >
            -
          </span>
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-12"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-12"
            />
          </svg>
        </span>
      </button>
      <select
        class="listbox-button__native"
        hidden=""
      >
        <option
          value="north-america"
        />
        <option
          value="south-america"
        />
      </select>
    </span>
  </div>
</div>
`;
